addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -310,7 +310,21 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                 this.currentOrder.updatePricelist(newClient);
             }
         }
-        _onClickPay() {
+        async _onClickPay() {
+            if (this.env.pos.get_order().orderlines.any(line => line.get_product().tracking !== 'none' && !line.has_valid_product_lot())) {
+                const { confirmed } = await this.showPopup('ConfirmPopup', {
+                    title: this.env._t('Warning'),
+                    body: this.env._t('You are trying to sell products with lot numbers, but some of them are not set'),
```

Comment:
I think it would be better to add a small summary of the issue in the title instead of `Warning` and add a question at the end of the body. (e.g. `Are you sure?`, `Do you want to proceed?` or `Continue without setting lot/serial number(s)?`)

Also, don't forget to add the translation keys in the `pot` file.